lib/spack/env/cc

```diff
@@ -162,7 +162,7 @@ fi
 # It doesn't work with -rpath.
 # This variable controls whether they are added.
 add_rpaths=true
-if [[ mode == ld && $OSTYPE == darwin* ]]; then
+if [[ $mode == ld && "$SPACK_SHORT_SPEC" =~ "darwin" ]]; then
```

Comment:
The example I found from the google search had the quotes.


________________________________
From: Erik Schnetter <notifications@github.com>
Sent: Thursday, April 21, 2016 9:32 PM
To: LLNL/spack
Cc: Patrick E Gartung; Author
Subject: Re: [LLNL/spack] remove use of unknown environment variable in lib/spack/env/cc (#821)


In lib/spack/env/cc<https://github.com/LLNL/spack/pull/821#discussion_r60684332>:

> @@ -162,7 +162,7 @@ fi
>  # It doesn't work with -rpath.
>  # This variable controls whether they are added.
>  add_rpaths=true
> -if [[ mode == ld && $OSTYPE == darwin* ]]; then
> +if [[ $mode == ld && "$SPACK_SHORT_SPEC" =~ "darwin" ]]; then


THIS IS PROBABLY WRONG. (Sorry for shouting, but I see that this has already been merged.) Quotes prevent the variable expansion here.

[[ ]] has very different quoting rules from [ ].

Given that the if condition probably never succeeded (as it was wrong before as well, missing $ sign), I wonder whether it is actually necessary?

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/LLNL/spack/pull/821/files/83eee4d4b0dd9dc69ed613eb8943ff9800304c8e#r60684332>